src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,109 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+
+/**
+ * Handles loading and saving of tasks to local storage.
+ */
+public class Storage {
+    private final Path HOME_DIRECTORY = Path.of(System.getProperty("user.dir") + "/data");
+    private File dataFile;
+    private final TaskList tasks;
+
+    public Storage(TaskList tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Loads tasks from a local file into a TaskList, creates one if one does not exist.
+     * @throws IOException If fail to load/create a storage file
+     */
+    public void load() throws IOException {
+        try {
+            if (!Files.exists(HOME_DIRECTORY)) {
+                Files.createDirectories(HOME_DIRECTORY);
+            }
+
+            this.dataFile = new File(this.HOME_DIRECTORY + "/duke.txt");
+            boolean isCreated = true;
+            if (!this.dataFile.exists()) {
+                isCreated = this.dataFile.createNewFile();
+            }
+            if (!isCreated) {
+                throw new DukeException("Failed to create storage file!");
+            }
+            Scanner sc = new Scanner(this.dataFile);
+            while (sc.hasNextLine()) {
+                String input = sc.nextLine();
+                parseFile(input, tasks);
+            }
+
+        } catch (FileNotFoundException | DukeException e) {
+            System.out.println(e);
+        }
+    }
+
+    /**
+     * Saves current TaskList into local file.
+     * @throws IOException If fail to save tasks into file
+     */
+    public void save() throws IOException {
+        try {
+            FileWriter fileWriter = new FileWriter(this.dataFile);
+            for (int i = 0; i < this.tasks.size(); i++) {
+                String line = tasks.get(i).toSaveFormat();
+                fileWriter.write(line);
+            }
+            fileWriter.close();
+        } catch (IOException e) {
+            System.out.println("Failed to save existing tasks...");
+        }
+    }
+
+    /**
+     * Parse input from file into tasks.
+     * @param input Line from local file
+     * @param tasks TaskList to store tasks
+     * @throws DukeException If fail to create task from invalid input
+     */
+    public static void parseFile(String input, TaskList tasks) throws DukeException {
+        String[] inputList = input.split(",");
+        String taskType = inputList[0];
+        String taskName = inputList[1];
+
+        switch (taskType) {
+            case "T":
+                tasks.addTask(new ToDo(taskName));
+                break;
+            case "D":
+                String by = inputList[2];
+                LocalDateTime deadline = LocalDateTime.parse(by, DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm"));
+                tasks.addTask(new Deadline(taskName, deadline));
+                break;
```

Comment:
Regarding variable declarations in `case` blocks, it might be better to wrap them inside their own scope. This prevents potential name clashes with other branches of the `switch` statement.

```suggestion
            case "D": {
                String by = inputList[2];
                LocalDateTime deadline = LocalDateTime.parse(by, DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm"));
                tasks.addTask(new Deadline(taskName, deadline));
                break;
            }
```

Reply:
Great observation, thanks! I wasn't sure of whether to add braces for every block as that would further increase the indentation of the code

resolved